python/ribasim/ribasim/model.py

```diff
@@ -133,11 +135,24 @@ def _save(self, directory: DirectoryPath, input_dir: DirectoryPath):
         for sub in self._nodes():
             sub._save(directory, input_dir)
 
+    def node_table(self) -> NodeTable:
+        """Compute the full NodeTable from all node types."""
+        df_chunks = [node.node.df for node in self._nodes()]
+        df = pd.concat(df_chunks, ignore_index=True)  # type: ignore
+        node_table = NodeTable(df=df)
+        node_table.sort()
+        return node_table
+
     def _nodes(self) -> Generator[MultiNodeModel, Any, None]:
```

Comment:
Is there really a reason to prefer a generator over an ordinary list here?

Reply:
Not really I guess. The implementation is slightly shorter. We were already using a generator for a similar function, and we only use this function to iterate over, so no `list(gen)`.